src/backend/utils/adt/timestamp.c

```diff
@@ -4170,12 +4170,11 @@ Datum
 timestamptz_floor(PG_FUNCTION_ARGS)
 {
 	TimestampTz timestamp = PG_GETARG_TIMESTAMPTZ(0);
-	Interval *span = PG_GETARG_INTERVAL_P(1);
-	fsec_t fsec;
-	struct pg_tm i_tm;
-	struct pg_tm ts_tm;
-	int tz;
-	int nset;
+	TimestampTz interval_ts;
+
+#ifndef HAVE_INT64_TIMESTAMP
```

Comment:
Figured this is kosher to do since all modern machines support 64-bit integers. If someone doesn't have them, then they're in bad luck.